app/models/container_template.rb

```diff
@@ -34,11 +34,23 @@ def instantiate(params, project = nil)
                                             :namespace => project
                                           },
                                           :objects    => objects,
-                                          :parameters => params)
+                                          :parameters => params_to_hashes(params))
     create_objects(processed_template['objects'], project)
     @created_objects.each { |obj| obj[:miq_class] = MIQ_ENTITY_MAPPING[obj[:kind]] }
   end
 
+  def params_to_hashes(params)
+    params.collect do |param|
+      {
+        :name     => param.name,
+        :value    => param.value,
+        :generate => param.generate,
+        :from     => param.from,
+        :required => param.required
+      }
+    end
+  end
+
```

Comment:
@cben Yes; I was going somewhere else in my brain before I realized `params` aren't just hashes themselves and are AR objects. 

Prefer `map` over `collect` ;)